FEAT (ansible): Ansible role that installs and runs fluentbit

cyb3r opened the pull request.

cyb3r committed:
FEAT (ansible): Ansible role that installs and runs fluentbit

andrewhowdencom commented on tasks/main.yml:
```diff
@@ -1,2 +1,20 @@
\ No newline at end of file
+- name: "Add an Apt signing key For Fluentbit"
+  apt_key:
+    url: "https://packages.fluentbit.io/fluentbit.key"
```
worth getting the key fingerprint here and adding that, so that if they change keys we know about it.

andrewhowdencom commented on tasks/main.yml:
```diff
@@ -1,2 +1,20 @@
+
+- name: "Add the Fluentbit repository into sources list"
+  apt_repository:
+    repo: "deb https://packages.fluentbit.io/ubuntu/xenial xenial main"
```
Instead of `xenial` you can use variable `{{ ansible_distribution_release }}`

andrewhowdencom commented on tasks/main.yml:
```diff
@@ -1,2 +1,20 @@
+    url: "https://packages.fluentbit.io/fluentbit.key"
+    state: present
+
+- name: "Add the Fluentbit repository into sources list"
```
Usually when writing the notices it's better to express them idempotently. Something like "Ensure the package repository is up to date"

andrewhowdencom commented on tasks/main.yml:
```diff
@@ -1,2 +1,20 @@
+
+- name: "Update repositories cache and install td-agent-bit"
+  apt:
+    name: "td-agent-bit"
```
https://fluentbit.io/documentation/0.8/installation/td-agent-bit.html

The more i know

andrewhowdencom commented on tasks/main.yml:
```diff
@@ -1,2 +1,20 @@
+- name: "Start service td-agent-bit, if not running"
+  service:
+    name: "td-agent-bit"
+    state: started
```
this should probably be "enabled"

cyb3r replied:
enabled: "yes" 
according to the docs for `enable`: `Whether the service should start on boot.`
it doesn't start it immediately as far as I understand. so having both might be better

cyb3r committed:
FEAT (ansible): ensures https transport is available

This commit ensures that https transport is available on the
remote machine before any other further calls which requires
HTTPs.

Also, it ensures that the `td-agent-bit` is started after boot.
This is done by introducing `enabled: "yes"`

cyb3r committed:
FEAT (ansible): updates the apt cache for apt-transport-https

cyb3r committed:
FEAT (ansible): adds the keyserver fingerprint to check against
